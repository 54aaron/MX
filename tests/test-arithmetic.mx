int main() {
    int i;
    float f;

    i = 9 + -8;
    print(i);

    i = 8 * (9 / 2);
    print(i);

    i = 3 / 2 + 1;
    print(i);

    f = 4.5 + 6;  # casting
    printf(f);

    f = 3 * 7.89;
    printf(f);  

    f = 11.0 / -2.75;
    printf(f);

    i = -1 / 3;
    print(i);

    i = -1  * -3;
    print(i);

    i = 1  * -3;
    print(i);

    f = -1.0 / -4.0;
    printf(f);

    f = 1.0 / -4.0;
    printf(f);
    
    f = 1.0 / 4.0;
    printf(f);

    f = -1.0 + -4.0;
    printf(f);
}
