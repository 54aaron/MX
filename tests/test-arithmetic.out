1
32
2
10.5
23.67
-4
0
3
-3
0.25
-0.25
0.25
-5